service/worker/indexer/indexer.go

```diff
@@ -249,7 +296,10 @@ func (i *Indexer) addMessageToES(indexMsg *indexer.Message, kafkaMsg messaging.M
 		return errUnknownMessageType
 	}
 
-	i.esProcessor.Add(req, keyToKafkaMsg, kafkaMsg)
+	i.primaryProcessor.Add(req, keyToKafkaMsg, kafkaMsg)
+	if i.secondaryProcessor != nil {
+		i.secondaryProcessor.Add(req, keyToKafkaMsg, kafkaMsg)
```

Comment:
Does it mean that there is no dynamic config flag controlling dual indexing?

Reply:
Yeah, only have a check when service start, if advanced == os2 and migration is enabled, we start the dual indexer which will set the secondaryProcessor. https://github.com/uber/cadence/pull/6284/files#diff-21d12a90c766bdd88cb02b14260e066a22d6f56a743e6575a5085f132b3323e7L220